### PR TITLE
Clean up and refactor project structure

### PR DESCRIPTION
**Pull request by leoliu49**
Addresses #14 

This current branch is branched off of #13 

`GUT*` --> `Universe` (e.g. UniverseChecker)
`GUTI*` --> `UniverseTypeInference` (e.g. UniverseTypeInferenceChecker)

The same has been applied to tests.
I haven't made any functional changes, and I'm currently working on #15

### CODE REVIEW

File: tests/typecheck/lostyes/LostYes.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package GUT.lostyes;
+package universe;
```
Comment (xingweitian) [inline tests/typecheck/lostyes/LostYes.java]: Should it be `typecheck.lostyes`?

File: tests/typecheck/strictpurity/Methods.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package GUT.strictpurity;
+package universe.strictpurity;
```
Comment (xingweitian) [inline tests/typecheck/strictpurity/Methods.java]: Same here.

File: tests/typecheck/topol/ArrayVP.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/ArrayVP.java]: Same here.

File: tests/typecheck/topol/ArrayWF.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/ArrayWF.java]: Same here.

File: tests/typecheck/topol/Boxing.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/Boxing.java]: Same here.

File: tests/typecheck/topol/CallEquals.java [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/CallEquals.java]: Same here.

File: tests/typecheck/topol/ComplexNew.java [hunk 7]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/ComplexNew.java]: Same here.

File: tests/typecheck/lostno/LostNo.java [hunk 8]
```diff
@@ -1,4 +1,4 @@
-package GUT.lostno;
+package typecheck.lostno;
```
Comment (xingweitian) [inline tests/typecheck/lostno/LostNo.java:1]: Here you write `typecheck.lostno`, while other places you use `universe`. Please unify the uses.

File: tests/typecheck/topol/CondExpr.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/CondExpr.java]: Same here.

File: tests/typecheck/topol/Defaults.java [hunk 10]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/Defaults.java]: Same here.

File: tests/typecheck/topol/FBound.java [hunk 11]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/FBound.java]: Same here.

File: tests/typecheck/topol/GenericInterfaces.java [hunk 12]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/GenericInterfaces.java]: Same here.

Comment (xingweitian) [reply tests/typecheck/topol/GenericInterfaces.java]: I found that `GBounds.java` does not indicate the package, should we add a package statement to it?

File: tests/typecheck/topol/Generics.java [hunk 13]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/Generics.java]: Same here.

File: tests/typecheck/topol/GenericsArrays.java [hunk 14]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/GenericsArrays.java]: Same here.

File: tests/typecheck/topol/MethodTypeVars.java [hunk 15]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/MethodTypeVars.java]: Same here.

File: tests/typecheck/topol/Wildcards.java [hunk 16]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/Wildcards.java]: Same here.

File: tests/typecheck/topol/StaticVP.java [hunk 17]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/StaticVP.java]: Same here.

File: tests/typecheck/topol/SimpleWF.java [hunk 18]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/SimpleWF.java]: Same here.

File: tests/typecheck/topol/SimpleVP.java [hunk 19]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/SimpleVP.java]: Same here.

File: tests/typecheck/topol/SimpleSub.java [hunk 20]
```diff
@@ -3,7 +3,7 @@
  * and open the template in the editor.
  */
 
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/SimpleSub.java]: Same here.

File: tests/typecheck/topol/SimpleNew.java [hunk 21]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/SimpleNew.java]: Same here.

File: tests/typecheck/topol/Methods.java [hunk 22]
```diff
@@ -1,4 +1,4 @@
-package GUT.topol;
+package universe.topol;
```
Comment (xingweitian) [inline tests/typecheck/topol/Methods.java]: Same here.

**Review by xingweitian — COMMENTED** [review]
Thanks a lot for this PR!

File: tests/typecheck/lostyes/LostYes.java (hunk 1 above)
Comment (leoliu49) [reply tests/typecheck/lostyes/LostYes.java]: Thanks for checking -- will fix for all the classes under `test`

**Comment by leoliu49** [comment]
Hi @xingweitian ! I've updated the tests to be consistent with `package typecheck.*`

**Review by xingweitian — APPROVED** [review]

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for this detailed work!
Can you do some more git magic to make it recognize that the test and VP files were moved and not deleted and added? For example, src/universe/UniverseViewpointAdaptor.java appears like a new file, but should be a renamed version of src/universe/GUTViewpointAdaptor.java. Same with a few test files.

I like the `GUT` to `Universe` renaming. For `GUTI`, the new name `UniverseTypeInference` is a bit odd. How about just `UniverseInference`?

**Comment by leoliu49** [comment]
> Can you do some more git magic to make it recognize that the test and VP files were moved and not deleted and added? For example, src/universe/UniverseViewpointAdaptor.java appears like a new file, but should be a renamed version of src/universe/GUTViewpointAdaptor.java. Same with a few test files.
> 

Hi Professor @wmdietl , `git` may not be recognizing some of the smaller renamings, and I'm having a little bit of trouble convincing them not to be delete + new file. I've tried separating the commits into a rename commit, then the actual refactoring, but tt seems that after squashing, `git` will just assume the files have been deleted-then-renamed anyway.
Is how it is right now okay?

**Review by wmdietl — APPROVED** [review]
Thanks for the renamings!

**Pull request merged**